tss/common/base.go

```diff
@@ -45,85 +46,75 @@ func (b *BaseTss) PopulatePartyStore(parties tss.SortedPartyIDs) {
 }
 
 // ProcessInboundMessages processes messages from tss parties and updates local party accordingly.
-func (b *BaseTss) ProcessInboundMessages(ctx context.Context, msgChan chan *comm.WrappedMessage) {
-	for {
-		select {
-		case wMsg := <-msgChan:
-			{
-				go func() {
-					defer func() {
-						if r := recover(); r != nil {
-							switch x := r.(type) {
-							case string:
-								b.ErrChn <- errors.New(x)
-							case error:
-								b.ErrChn <- x
-							default:
-								b.ErrChn <- errors.New("unknown panic")
-							}
-						}
-					}()
+func (b *BaseTss) ProcessInboundMessages(p *pool.ContextPool, msgChan chan *comm.WrappedMessage) {
+	p.Go(func(ctx context.Context) error {
+		for {
+			select {
+			case wMsg := <-msgChan:
+				{
 					b.Log.Debug().Msgf("processed inbound message from %s", wMsg.From)
 
 					msg, err := UnmarshalTssMessage(wMsg.Payload)
 					if err != nil {
-						b.ErrChn <- err
-						return
+						return err
 					}
 
-					ok, err := b.Party.UpdateFromBytes(
-						msg.MsgBytes,
-						b.PartyStore[wMsg.From.Pretty()],
-						msg.IsBroadcast,
-						new(big.Int).SetBytes([]byte(b.SID)))
-					if !ok {
-						b.ErrChn <- err
+					var pc panics.Catcher
+					pc.Try(func() {
+						ok, err := b.Party.UpdateFromBytes(
+							msg.MsgBytes,
+							b.PartyStore[wMsg.From.Pretty()],
+							msg.IsBroadcast,
+							new(big.Int).SetBytes([]byte(b.SID)))
+						if !ok {
+							panic(err)
+						}
+					})
```

Comment:
I think I just blanked and forgot to do it properly. 
As I remember I made this because panic catcher expects another function from which you can't get the result or return an error. 

The tss-lib sometimes can panic if someone sends an invalid format message (that is why I want to catch a panic).